lib/src/api.rs

```diff
@@ -185,83 +82,50 @@ impl Api {
     /// # Examples
     ///
     /// ```rust
-    /// # extern crate futures;
-    /// # extern crate telegram_bot;
-    /// # extern crate tokio_core;
-    /// # use futures::Future;
+    /// #![feature(async_await)]
     /// # use telegram_bot::{Api, GetMe};
-    /// # use tokio_core::reactor::Core;
+    /// # use std::time::Duration;
     /// #
-    /// # fn main() {
-    /// # let core = Core::new().unwrap();
+    /// # #[tokio::main]
+    /// # async fn main() {
     /// # let telegram_token = "token";
-    /// # let api = Api::configure(telegram_token).build(core.handle()).unwrap();
+    /// # let api = Api::new(telegram_token);
     /// # if false {
-    /// use std::time::Duration;
-    ///
-    /// let future = api.send_timeout(GetMe, Duration::from_secs(5));
-    /// future.and_then(|me| Ok(assert!(me.is_some())));
+    /// let result = api.send_timeout(GetMe, Duration::from_secs(2)).await;
+    /// println!("{:?}", result);
     /// # }
     /// # }
     /// ```
-    pub fn send_timeout<Req: Request>(
+    pub async fn send_timeout<Req: Request>(
         &self,
         request: Req,
         duration: Duration,
-    ) -> TelegramFuture<Option<<Req::Response as ResponseType>::Type>> {
-        let timeout_future = result(Timeout::new(duration, &self.inner.handle))
-            .flatten()
-            .map_err(From::from)
-            .map(|()| None);
-        let send_future = self.send(request).map(|resp| Some(resp));
-
-        let future = timeout_future
-            .select(send_future)
-            .map(|(item, _next)| item)
-            .map_err(|(item, _next)| item);
-
-        TelegramFuture::new(Box::new(future))
+    ) -> Result<<Req::Response as ResponseType>::Type, Error> {
+        send_timeout_request(self.token.clone(), request, duration).await
```

Comment:
Wouldn't it be better for `send_timeout_request` and `send_request` to receive a token as `&str`? It would avoid those `.clone()` calls